### PR TITLE
Update deps and uncomment test

### PR DESCRIPTION
**Pull request by smaldini**
Resolve https://jira.spring.io/browse/INT-3644 and update reactor dependencies

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -264,6 +264,7 @@ project('spring-integration-core') {
 		compile "org.springframework:spring-tx:$springVersion"
 		compile "org.springframework.retry:spring-retry:$springRetryVersion"
 		compile ("io.projectreactor:reactor-core:$reactorVersion", optional)
+		compile ("io.projectreactor:reactor-stream:$reactorVersion", optional)
```
Comment (artembilan) [inline build.gradle]: After this, it looks redundant to have `reactor-core` here.
Isn't it?
BTW, having you change in `master` we won't be able to merge your work properly

File: spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java [hunk 2]
```diff
@@ -16,42 +16,33 @@
...
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
```
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java]: We don't like this asterisk, but it is minor and will be fixed on merge

File: build.gradle (hunk 1 above)
Comment (smaldini) [inline build.gradle]: Right I have some bad habit to make explicit deps you're right!

File: spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java (hunk 2 above)
Comment (smaldini) [inline spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java]: (y)

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/78a8d713a3dc8fdb6c5df29d9c161274879ac4a8

**Pull request closed without merging**